lib/matplotlib/bezier.py

```diff
@@ -177,18 +225,70 @@ class BezierSegment:
     """
 
     def __init__(self, control_points):
-        n = len(control_points)
-        self._orders = np.arange(n)
-        coeff = [math.factorial(n - 1)
-                 // (math.factorial(i) * math.factorial(n - 1 - i))
-                 for i in range(n)]
-        self._px = np.asarray(control_points).T * coeff
+        self.cpoints = np.asarray(control_points)
+        self.n, self.d = self.cpoints.shape
```

Comment:
let's not introduce new public API unless we really need to.

Reply:
good idea. how about a read-only property instead?

Reply:
n and d should definitely be readonly properties.
I guess I would name them `.control_points`, `.degree` (which is n-1!) and `.dimension` (we may as well be explicit).

Reply:
I left `.control_points` read-only for now. Currently nothing in the class assumes that the control points don't change (except n and d). But figured there's no reason to allow it anyway. The user can always create a separate BezierSegment object with new control points.